src/main/java/com/google/sps/data/Cache.java

```diff
@@ -1,90 +1,77 @@
 package com.google.sps.data;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 
 /**
  * This class coordinates retrieving information from the data sources needed by
  * the servlets. This includes filtering the data from the data sources into
  * only that needed by the servlet and collating the data from multiple sources.
  */
 public class Cache {
-  private GriddedData dataGrid;
+  private HashMap<Integer,HashMap<Integer,GridCell>> dataGrid = new HashMap<>();
 
-  public GriddedData getGrid(int zoomLevel, Coordinates swCorner, Coordinates neCorner) {
+  public GriddedData getGrid(Coordinates swCorner, Coordinates neCorner) {
```

Comment:
When this method is called multiple times it appears it'll fetch the NSWGovData, and add those point to the existing cells, again. Is this the intended behaviour?

Reply:
At the moment, yes. The goal would be to check when it was last updated before updating again, but for now I wanted to focus on getting something working even if it fetches every time

Reply:
My concern is more around that it doesn't appear to re-initialise `dataGrid` between each call; so consecutive calls will be the resulting of averaging in each new calls values with the current grid state.

If the intent is to simply not cache; then ensure the dataGrid is cleared between each fetch, after the new data is available to populate it. (ie, don't invalidate the cache too early, so it's still available of `getNSWGovData()` raises an exception).